modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,333 @@
+styles work and will be explained more in depth below.
+
+<ExampleCodeBlock code={ResponsiveContainer} />
+
+### Applying Container-Based Responsive Styling
+
+For container-based responsive styling, we have utilities that will allow you to apply responsive
+styles at those breakpoints. There are four utilities/hooks:
+`useResizeObserver | ResponsiveContextProvider | useResponsiveContext | useResponsiveStyles`
+
+`useResizeObserver` - This is a hook that will provide the width of the container. This can used to
+measure container size as well.
+
+`ResponsiveContextProvider` - This React Context Provider that will accept the width of the
+container from `useResizeObserver`.
+
+`useResponsiveContext` - This hook subscribes to the `ResponsiveContextProvider` and will determine
+whether the current width provided from `useResizeObserver` is within a particular theme breakpoint.
+
+`useResponsiveContainerStyles` - This is a hook that will allow you to create container-based
+responsive styles with objects (as you can see in the example below). This hook accepts three
+arguments with the third being optional (Style Objects, Container Width, Theme). In each style
+object, there are five sizes that it will accept: `zero, small, medium, large and extra large`.
+These give sizes represent the breakpoints that are at the bottom of this page. The sizes will act
+like `min-width`. For example, if you want to apply styles from `medium` and up, then you would
+write those styles under `m`.
+
+In the example below, we are importing three of those utilities:
+`useResizeObserver | ResponsiveContextProvider | useResponsiveStyles`
+
+See example below:
+
+```tsx
+import {Flex, Box} from '@workday/canvas-kit-react/layout';
+import {
+  useResizeObserver,
+  ResponsiveContextProvider,
+  useResponsiveContainerStyles,
+} from '@workday/canvas-kit-react/common';
+
+const ref = React.useRef(null);
+const [width, setWidth] = React.useState(0);
+
+useResizeObserver({
+  ref: ref,
+  onResize: data => {
+    setWidth(data.width || 0);
+  },
+});
+
+const containerResponsiveStyles = useResponsiveContainerStyles(
+  {
+    flex: {
+      flexDirection: 'column',
+      padding: 'm',
+      depth: 1,
+      borderRadius: 'l',
+      zero: {
+        backgroundColor: 'Red',
+      },
+      s: {
+        backgroundColor: 'Orange',
+      },
+      m: {
+        backgroundColor: 'Yellow',
+      },
+      l: {
+        backgroundColor: 'Green',
+      },
+      xl: {
+        backgroundColor: 'Blue',
+      },
+    },
+    box: {
+      padding: 's',
+    },
+  },
+  width
+);
+
+return (
+  <ResponsiveContextProvider width={width}>
+    <Box ref={ref}>
+      <Flex {...containerResponsiveStyles.flex}>
+        <Box {...containerResponsiveStyles.box}>Hello World</Box>
+      </Flex>
+    </Box>
+  </ResponsiveContextProvider>
+);
+```
+
+### Viewport Based Responsive Styling
+
+The viewport is the user's visible area of a web page and it varies with the device.
```

Comment:
again should we say device or screen size?

Reply:
Screen size for sure